test_timer failure

I've been seeing `test_timer` fail locally for a long time and noticed this test has a few conditional skips which makes me think it needs to be skipped for my system -- if so, I'd like to implement that skip but need some help tracking down the issue.

The failure:

```
_____________________________________________________________ test_timer _____________________________________________________________

    @pytest.mark.skipif(
        platform.system() == 'Darwin',
        reason='vtkCocoaRenderWindowInteractor (MacOS) does not invoke TimerEvents during ProcessEvents. ',
    )
    @pytest.mark.needs_vtk_version(
        (9, 2),
        reason='vtkXRenderWindowInteractor (Linux) does not invoke TimerEvents during ProcessEvents until VTK9.2.',
    )
    def test_timer():
        # Create a normal interactor from the offscreen plotter (not generic,
        # which is the default for offscreen rendering)
        pl = pyvista.Plotter()
        iren = pyvista.plotting.render_window_interactor.RenderWindowInteractor(pl)
        iren.set_render_window(pl.render_window)
    
        duration = 50  # Duration of created timers
        delay = 5 * duration  # Extra time we wait for the timers to fire at least once
        events = []
    
        def on_timer(obj, event):
            # TimerEvent callback
            events.append(event)
    
        def process_events(iren, duration):
            # Helper function to call process_events for the given duration (in milliseconds).
            t = 1000 * time.time()
            while 1000 * time.time() - t < duration:
                iren.process_events()
    
        # Setup interactor
        iren.add_observer("TimerEvent", on_timer)
        iren.initialize()
    
        # Test one-shot timer (only fired once for the extended duration)
        iren.create_timer(duration, repeating=False)
        process_events(iren, delay)
>       assert len(events) == 1
E       assert 0 == 1
E        +  where 0 = len([])

tests/plotting/test_render_window_interactor.py:169: AssertionError
```

My system

```

--------------------------------------------------------------------------------
  Date: Sat Jun 10 17:06:20 2023 EDT

                  OS : Linux
              CPU(s) : 20
             Machine : x86_64
        Architecture : 64bit
                 RAM : 93.0 GiB
         Environment : Python
         File system : ext4
          GPU Vendor : NVIDIA Corporation
        GPU Renderer : Quadro RTX 5000/PCIe/SSE2
         GPU Version : 4.6.0 NVIDIA 525.85.12
    MathText Support : False

  Python 3.9.15 (main, Dec  1 2022, 16:54:05)  [GCC 9.4.0]

             pyvista : 0.40.dev0
                 vtk : 9.2.5
               numpy : 1.23.5
          matplotlib : 3.6.3
              scooby : 0.7.1
               pooch : v1.6.0
             imageio : 2.22.4
             IPython : 8.7.0
            colorcet : 3.0.1
             cmocean : 2.0
          ipyvtklink : 0.2.3
          ipywidgets : 8.0.6
               scipy : 1.10.0
                tqdm : 4.64.1
              meshio : 5.3.4
          jupyterlab : 3.5.2
           pythreejs : 2.4.1
      pytest_pyvista : 0.2.dev0
               trame : 2.3.2
        trame_client : 2.7.5
        trame_server : 2.10.0
           trame_vtk : 2.4.1
jupyter_server_proxy : 3.2.2
        nest_asyncio : 1.5.6
--------------------------------------------------------------------------------
```

I have a custom build of VTK with EGL offscreen rendering

--- a/tests/plotting/test_render_window_interactor.py
+++ b/tests/plotting/test_render_window_interactor.py
@@ -130,7 +130,7 @@ def test_track_click_position():
 
 
 @pytest.mark.skipif(
-    not isinstance(_vtk.vtkRenderWindowInteractor(), (_vtk.vtkWin32RenderWindowInteractor, _vtk.vtkXRenderWindowInteractor)),
+    type(_vtk.vtkRenderWindowInteractor()).__name__ not in ("vtkWin32RenderWindowInteractor", "vtkXRenderWindowInteractor"),
     reason='Other RenderWindowInteractors do not invoke TimerEvents during ProcessEvents.',
 )
 @pytest.mark.needs_vtk_version(

--- a/tests/plotting/test_render_window_interactor.py
+++ b/tests/plotting/test_render_window_interactor.py
@@ -1,6 +1,5 @@
 """Test render window interactor"""
 
-import platform
 import time
 
 import pytest
@@ -131,8 +130,8 @@ def test_track_click_position():
 
 
 @pytest.mark.skipif(
-    platform.system() == 'Darwin',
-    reason='vtkCocoaRenderWindowInteractor (MacOS) does not invoke TimerEvents during ProcessEvents. ',
+    not isinstance(_vtk.vtkRenderWindowInteractor(), (_vtk.vtkWin32RenderWindowInteractor, _vtk.vtkXRenderWindowInteractor)),
+    reason='Other RenderWindowInteractors do not invoke TimerEvents during ProcessEvents.',
 )
 @pytest.mark.needs_vtk_version(
     (9, 2),

--- a/tests/plotting/test_render_window_interactor.py
+++ b/tests/plotting/test_render_window_interactor.py
@@ -130,7 +130,8 @@ def test_track_click_position():
 
 
 @pytest.mark.skipif(
-    type(_vtk.vtkRenderWindowInteractor()).__name__ not in ("vtkWin32RenderWindowInteractor", "vtkXRenderWindowInteractor"),
+    type(_vtk.vtkRenderWindowInteractor()).__name__
+    not in ("vtkWin32RenderWindowInteractor", "vtkXRenderWindowInteractor"),
     reason='Other RenderWindowInteractors do not invoke TimerEvents during ProcessEvents.',
 )
 @pytest.mark.needs_vtk_version(

